qutip/core/data/convert.pyx

```diff
@@ -259,32 +261,64 @@ cdef class _to:
                     _converter(convert[::-1], to_t, from_t)
         for dispatcher in self.dispatchers:
             dispatcher.rebuild_lookup()
+        for dtype in self.dtypes:
+            self._str2type[dtype.__name__.lower()] = dtype
+
+    def parse(self, dtype):
+        """
```

Comment:
@Ericgig I updated the docstring and the error messages. Let me know what you think.

Reply:
Do we expect user to know this function exist?
If so, do we want to use the `data-layer` term?
data-layer is a new term for users that we use internally, not sure if we describe it anywhere.

If we don't, It's great.

Reply:
I think we don't generally expect users to know it exists, but it'll show up if they do `help(data.to)` or look in the API documentation, so we should make sure its docstring is clear.

I think if we commit to the aliases just being strings, it'll be fine to call the type `type or str` all the way through the codebase.